lib/shopify-cli/theme/development_theme.rb

```diff
@@ -58,7 +69,10 @@ def generate_theme_name
         hostname = Socket.gethostname.split(".").shift
         hash = SecureRandom.hex(3)
 
-        theme_name = "Development (#{hash}-#{hostname})"
+        theme_name = "Development ()"
+        hostname_character_limit = API_NAME_LIMIT - theme_name.length - hash.length - 1
+        identifier = "#{hash}-#{hostname.truncate(hostname_character_limit)}"
```

Comment:
I don't think you need to bring in that ActiveSupport method, you can use `[]` instead:

```suggestion
        identifier = "#{hash}-#{hostname[0, hostname_character_limit]}"
```